### ISSUE TITLE
BUILD FAILING on various configurations

### ISSUE THREAD

**Issue by parndt** [rubysherpas/forem#456]
See [the CI builds](https://travis-ci.org/radar/forem/builds)

**Comment by daniely** [comment]
failing [ruby 2.0](https://travis-ci.org/radar/forem/jobs/10170036) and [1.9.3](https://travis-ci.org/radar/forem/jobs/10170039) builds are in **controller** tests that use `Warden::Test::Helpers`. according to [stackoverflow](http://stackoverflow.com/a/17050993/600953) this isn't even supported.

> Warden is a Rack middleware, but RSpec controller specs don't even include Rack, as these types of specs are not meant to run your full application stack, but only your controller code.

although it works in some environments it looks like a happy accident. while there are [workarounds](http://kentaroimai.com/articles/1-controller-test-helpers-for-warden) IMHO the correct solution is to move these tests into request/integration tests. i'd be happy to work on it (making it a priority over adding cache_counters)

as for [jruby](https://travis-ci.org/radar/forem/jobs/10170042) builds i'm guessing it's something to do with `moderator_ids` but i'm not 100% sure on that. will look into it after fixing those controller tests.

**Comment by parndt** [comment]
Legend!

**Comment by daniely** [comment]
@parndt any ideas on how to write a **request** spec to create a post reply to a topic?

i asked on [stackoverflow](http://stackoverflow.com/questions/18317707/route-not-found-in-rails-engine-rspec-request-spec) because i've been stuck on this for hours.

**Comment by parndt** [comment]
@daniely what about like [the one we have](https://github.com/radar/forem/blob/master/spec/features/posts_spec.rb#L45-L51)?

**Comment by daniely** [comment]
@parndt i consider that a feature spec. i'm using the definition of request vs feature spec found [here](http://www.andylindeman.com/2012/11/11/rspec-rails-and-capybara-2.0-what-you-need-to-know.html)

one case i'm testing is disallowing a user from constructing a POST request (say via curl) to reply to a topic in a forum they are not allowed to access. this was possible before one of my PRs which had a small permissions fix.

at this point i don't think it's worth all the effort but and maybe something like a cancan test or feature spec will suffice. but i'm still curious about how to get a working request spec.

**Comment by parndt** [comment]
As far as I'm aware rspec deprecated request specs in favour of feature specs.. is that inaccurate?

**Comment by daniely** [comment]
@parndt i don't think it's deprecated since there's [rspec 2.14 documentation](https://www.relishapp.com/rspec/rspec-rails/docs/request-specs/request-spec) for it and it's mentioned in [rspec-rails readme](https://github.com/rspec/rspec-rails/blob/master/README.md#request-specs)

**Comment by radar** [comment]
@daniely I would tackle that test using a controller spec, not a request spec. You can fake a POST request to a controller that way.

**Comment by daniely** [comment]
@radar yep, due to the routing problem i was having i ended up converting it back to a controller spec. the build was fixed by not using the warden `sign_in` helper in controllers.

**Comment by radar** [comment]
Goodo. Thanks for your work on this.
